lib/deploy.js

```diff
@@ -32,7 +32,19 @@ module.exports = promisify(function(config, callback) {
   var ghPagesConfig = {};
   var rootDir = config.rootDir ? config.rootDir : '.';
   var cloneDir = config.cloneDir ? config.cloneDir : null;
-  var commitMessage = config.message ? config.message : childProcess.execSync('git log --format=%B -n 1').toString().trim();
+
+  var commitMessage = config.message || (function() {
+    var spawn = childProcess.spawnSync('git', ['log', '--format=%B', '-n', '1']);
+    var errorText = spawn.stderr.toString();
+
+    if (errorText) {
+      gutil.log('Fatal error from `git log`.  You must have one commit before deploying.  Error detail: ' + errorText);
+      throw new Error();
```

Comment:
Nit: I'd make the raw *errorText* be the Error object's *message* instead of appending it to the friendly message, i.e. instantiate the Error object with ```new Error(errorText)```.

Then, instead of seeing:

>[18:15:23] Fatal error from `git log`.  You must have one commit before deploying.  Error detail: fatal: bad default revision 'HEAD'
>[Error]

A developer will see this:

>[18:17:41] Fatal error from `git log`.  You must have one commit before deploying.
>[Error: fatal: bad default revision 'HEAD']

Which feels a bit more readable.